tests/qadence/test_error_models.py

```diff
@@ -0,0 +1,227 @@
+from __future__ import annotations
+
+from collections import Counter
+
+import numpy as np
+import pytest
+import torch
+from numpy.random import rand
+from sympy import acos
+
+import qadence as qd
+from qadence import BackendName
+from qadence.blocks import (
+    AbstractBlock,
+    add,
+    kron,
+)
+from qadence.circuit import QuantumCircuit
+from qadence.constructors.hamiltonians import hamiltonian_factory
+from qadence.divergences import js_divergence
+from qadence.measurements.protocols import Measurements
+from qadence.models import QuantumModel
+from qadence.noise import Noise
+from qadence.noise.readout import WhiteNoise, bs_corruption, create_noise_matrix, sample_to_matrix
+from qadence.operations import (
+    CNOT,
+    RX,
+    RZ,
+    H,
+    HamEvo,
+    X,
+    Y,
+    Z,
+)
+from qadence.types import DiffMode
+
+
+@pytest.mark.parametrize(
+    "error_probability, counters, exp_corrupted_counters, n_qubits",
+    [
+        (
+            1.0,
+            [Counter({"00": 27, "01": 23, "10": 24, "11": 26})],
+            [Counter({"11": 27, "10": 23, "01": 24, "00": 26})],
+            2,
+        ),
+        (
+            1.0,
+            [Counter({"001": 27, "010": 23, "101": 24, "110": 26})],
+            [Counter({"110": 27, "101": 23, "010": 24, "001": 26})],
+            3,
+        ),
+    ],
+)
+def test_bitstring_corruption_all_bitflips(
+    error_probability: float, counters: list, exp_corrupted_counters: list, n_qubits: int
+) -> None:
+    n_shots = 100
+    noise_matrix = create_noise_matrix(WhiteNoise.UNIFORM, n_shots, n_qubits)
+    err_idx = np.array([(item).numpy() for i, item in enumerate(noise_matrix < error_probability)])
+    sample = sample_to_matrix(counters[0])
```

Comment:
are these tests deterministic? should we set a seed/make the test flaky to make sure its not sometimes failing?

Reply:
This particular test is deterministic in a way as the probability to invert the bit strings is 1.0, so we know what to expect.